pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch

```diff
@@ -634,808 +628,67 @@ index aba87d5..c059568 100644
  
  /*==============================================================================
                                       MACROS
-diff --git a/utils/inc/clist.h b/utils/inc/clist.h
-deleted file mode 100644
-index 0e26be8..0000000
+diff --git a/utils/inc/clist.h b/utils/inc/emb6_clist.h
+similarity index 94%
+rename from utils/inc/clist.h
+rename to utils/inc/emb6_clist.h
+index 0e26be8d..a104f29e 100644
 --- a/utils/inc/clist.h
-+++ /dev/null
-@@ -1,163 +0,0 @@
--/*
-- * Copyright (c) 2004, Swedish Institute of Computer Science.
-- * All rights reserved.
-- *
-- * Redistribution and use in source and binary forms, with or without
-- * modification, are permitted provided that the following conditions
-- * are met:
-- * 1. Redistributions of source code must retain the above copyright
-- *    notice, this list of conditions and the following disclaimer.
-- * 2. Redistributions in binary form must reproduce the above copyright
-- *    notice, this list of conditions and the following disclaimer in the
-- *    documentation and/or other materials provided with the distribution.
-- * 3. Neither the name of the Institute nor the names of its contributors
-- *    may be used to endorse or promote products derived from this software
-- *    without specific prior written permission.
-- *
-- * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
-- * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
-- * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
-- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-- * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
-- * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
-- * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-- * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
-- * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
-- * SUCH DAMAGE.
-- *
-- * This file is part of the Contiki operating system.
-- *
-- * Author: Adam Dunkels <adam@sics.se>
-- *
```

Comment:
This changes the content of the `emb6` package and has AFAIK nothing to do with the git version change. Is this intended?

Reply:
nope, my work flow was as follows: 

- checkout emb6
- apply patches from `RIOT/pkg/emb6/patches/` using an _older_ git version
- (re)generate patches using _newer_ git version (which fails patching) with `git format-patch -n 14e4a3cfff01644e078870e14e16a1fe60dcc895`

I did not touch any files

Reply:
looks like the _newer_ git version also generates patches differently, not just hash length and whitespace 🤔

Reply:
```diff
-deleted file mode 100644
...
-new file mode 100644
```
The old git version removes and adds this file, while the new git version correctly identifies that as a rename
```diff
+rename to utils/inc/emb6_clist.h
```

Reply:
ah, now I get it, thx @cgundogan - so the newer git version is smarter by simply renaming instead of del+add. So @miri64 looks like the new patches do not change the package content after all.